rust/src/data/ffi/mod.rs

```diff
@@ -246,8 +248,21 @@ pub extern "C" fn opendp_data__slice_as_object(
         // Error checking based on pyo3-polars:
         // https://github.com/pola-rs/pyo3-polars/blob/5150d4ca27c287ff4be5cafef243d9a878a8879d/pyo3-polars/src/lib.rs#L147-L153
         // the slice is lf.__getstate__ from the python side and then deserialized here
-        ciborium::de::from_reader(slice).map_err(
-            |e| err!(FFI, "Error when deserializing {}. This may be because you're using features from Polars that are not currently supported. {}", name, e)
+        polars_utils::pl_serialize::SerializeOptions::default()
+            // `false` disables forward compatibility
+            .deserialize_from_reader::<_, _, false>(slice)
+            .map_err(
+            |e| err!(FFI, "Error when deserializing '{}'. This may be because you're using features from Polars that are not currently supported. {}", name, e)
+        )
+    }
+    #[cfg(feature = "polars")]
+    pub fn deserialize_raw_dslplan(
+        raw: &FfiSlice
+    ) -> Fallible<DslPlan> {
+        let slice = unsafe { slice::from_raw_parts(raw.ptr as *const u8, raw.len) };
+        DslPlan::deserialize_versioned(slice)
+            .map_err(
+            |e| err!(FFI, "Error when deserializing 'DslPlan'. This may be because you're using features from Polars that are not currently supported. {}", e)
```

Comment:
Many serializations include a version number. Any chance we have that information so we could check and have an error message like this?
> Was expecting polars vX but got polars vY.

(Might even check the version number first.)

Reply:
Version mismatch should be caught separately, leaving only unsupported features.